### ISSUE TITLE
Useful context menu that includes "Reopen Last Closed Tab" should be available across entire tab bar

### ISSUE THREAD

**Issue by philkloose** [brave/browser-laptop#8303]
- Did you search for similar issues before submitting this one? Yes

- Describe the issue you encountered: I'm constantly right-clicking on the tab bar (to the _right_ of the open tabs to try and reopen last closed tab.  At the moment, we get a context menu that doesn't seem to fit the context.  In opinion, the option to reopen closed tabs should be available from here and possibly even more options.

- Platform (Win7, 8, 10? macOS? Linux distro?):
Win 7
- Brave Version (revision SHA):
 
Brave: 0.14.1 
rev: 3de60d52df3ef48d12d9696a076b72079beb9658 
Muon: 2.57.8 
libchromiumcontent: 57.0.2987.133 
V8: 5.7.492.71 
Node.js: 7.4.0 
Update Channel: dev 
os.platform: win32 
os.release: 6.1.7601 
os.arch: x64

- Steps to reproduce:
    1. Open a tab.
    2. Close the tab.
    3. Right-click to the right of your open tabs to get a context menu and notice that the options don't make sense contextually.
    
- Actual result:
![no-reopenlasttab](https://cloud.githubusercontent.com/assets/5502734/25018365/b79a4e8c-204c-11e7-94e4-bb4d1dfe6225.png)
- Expected result:
![image](https://cloud.githubusercontent.com/assets/4733304/25409303/3d497bca-29c6-11e7-9c6a-6ee58a8cac76.png)

- Will the steps above reproduce in a fresh profile? If not what other info can be added?
Yes
- Is this an issue in the currently released version?
Yes
- Can this issue be consistently reproduced?
Yes

**Comment by philkloose** [comment]
I attempted to implement this. 
#8320

**Comment by srirambv** [comment]
This is not reliable on Windows. Out of 20-30 right clicks I got the expected menu once.  Right click outside the new tab button shows window context menu. Right click on the new tab shows new tab context menu but second right click shows the same window context menu 
![8303 1](https://cloud.githubusercontent.com/assets/17010094/25662932/32a79e56-300e-11e7-8239-b775f5a2b874.gif)

**Comment by philkloose** [comment]
My experience is similar but I have since trained myself only to click at the bottom of the tab toolbar which works 100% of the time for me.  I've always chalked it up to the more generalized problem of the hitboxes (not sure what else to call them) being off.  For me, in Windows 7, it happens in the other toolbars (URL, Bookmarks, Tabs) too.  In the URL bar, I have to click in the upper half or it doesn't allow me to edit text.  The problems are exacerbated if my window is maximized.  I don't have near the requisite knowledge to start fixing this, but I think it's related to issue #7641.



In the

**Comment by philkloose** [comment]
@srirambv Hmm, I just built the latest preview and it appears you're right.  The hitbox is even smaller than before when I was testing it.  I have no idea why it's so off, but you need to click in approx the bottom 5px of the tab bar to trigger it properly.

**Comment by bsclifton** [comment]
@philkloose the milestone was removed from this one because the change ended up being reverted

Definitely my fault for not testing on macOS before merging ☹️   However, if you wanted to try this one again, you can cherry pick in the following two commits into a branch:
```
git checkout master
git pull
git checkout -b add-reopen-menu
git cherry-pick e2706d795670c9ec894e3030029b8affd1b56e4a
git cherry-pick 6ebb6d29e1c56cf24ed3af66ac3aa77ac784555d
```
At that point, you'll have all the code which was in your PR.  Then, the remaining two tasks are:
1. Ensure right clicking anywhere in the tab toolbar works for Windows
2. Ensure menu isn't shown after right clicking the + button in the tab toolbar on macOS

**Comment by bsclifton** [comment]
Fixing this properly would require either making the tabs bar non-draggable OR solving https://github.com/brave/browser-laptop/issues/4244

**Comment by bsclifton** [comment]
Blocked by https://github.com/brave/browser-laptop/issues/4244

**Comment by eljuno** [comment]
+1 from community https://community.brave.com/t/re-opening-closed-tab-and-related-options-are-not-present-while-clicking-on-the-tabs-bar-only-available-while-right-clicking-on-the-currently-open-tabs/26734?u=eljuno

**Comment by bsclifton** [comment]
Closing as `wontfix` since we won't be able to address https://github.com/brave/browser-laptop/issues/4244, which blocks this on Windows.  This is resolved in [brave-core](https://github.com/brave/brave-browser)
![screen shot 2018-08-06 at 5 51 24 pm](https://user-images.githubusercontent.com/4733304/43747893-639b11ac-99a1-11e8-9fda-4c6077a551d0.png)
